server/src/main/java/org/opensearch/search/aggregations/AggregatorBase.java

```diff
@@ -236,6 +242,29 @@ protected boolean tryPrecomputeAggregationForLeaf(LeafReaderContext ctx) throws
         return false;
     }
 
+    /**
+     * Returns true if currently in precompute path
+     * @return
+     */
+    protected boolean isTryPrecomputePath() {
+        if (precomputePath) {
+            return true;
+        } else if (parent == null) {
+            return false;
+        }
+        Aggregator current = parent;
+        do {
+            if (current instanceof AggregatorBase base) {
+                if (base.precomputePath) {
+                    return true;
+                }
+            }
+            current = current.parent();
+        } while (current != null);
+
+        return precomputePath;
+    }
```

Comment:
I think this way makes sense. But I would suggest to make the logic tight to what we need.

I think what we need here is for the sub agg to know whether its parent is doing filter rewrite optimization. And only the direct parent, not to iteratively go up the parent chain.